pkg/codegen/python/python_test.go

```diff
@@ -0,0 +1,40 @@
+package python
+
+import "testing"
+
+var pyNameTests = []struct {
+	input    string
+	expected string
+	legacy   string
+}{
+	{"kubeletConfigKey", "kubelet_config_key", "kubelet_config_key"},
+	{"podCIDR", "pod_cidr", "pod_cidr"},
+	{"podCidr", "pod_cidr", "pod_cidr"},
+	{"podCIDRs", "pod_cidrs", "pod_cid_rs"},
+	{"someTHINGsAREWeird", "some_things_are_weird", "some_thin_gs_are_weird"},
+	{"podCIDRSet", "pod_cidr_set", "pod_cidr_set"},
+	{"Sha256Hash", "sha256_hash", "sha256_hash"},
+	{"SHA256Hash", "sha256_hash", "sha256_hash"},
+}
+
+func TestPyName(t *testing.T) {
+	for _, tt := range pyNameTests {
+		t.Run(tt.input, func(t *testing.T) {
+			result := PyName(tt.input)
+			if result != tt.expected {
+				t.Errorf("expected \"%s\"; got \"%s\"", tt.expected, result)
+			}
```

Comment:
Nit: You could import `"github.com/stretchr/testify/assert"` and do:

```suggestion
			assert.Equal(t, tt.expected, result)
```